sympy/core/function.py

```diff
@@ -2834,9 +2834,16 @@ def expand_log(expr, deep=True, force=False):
     (x + y)*(log(x) + 2*log(y))*exp(x + y)
 
     """
+    from sympy import Mul, log
+    if factor is False:
+        expr = expr.replace(
+        lambda x: x.is_Mul and all(any(isinstance(i, log) and i.args[0].is_Rational
+        for i in Mul.make_args(j)) for j in x.as_numer_denom()),
+        lambda x: expand(x, factor=True))
```

Comment:
Since this is just the `expand_log` hint, we probably shouldn't apply the general `expand`, just `expand_log`